metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/bolt/ParserBolt.java

```diff
@@ -59,10 +61,22 @@ public ParserBolt withMessageFilter(MessageFilter<JSONObject> filter) {
   public void prepare(Map stormConf, TopologyContext context, OutputCollector collector) {
     super.prepare(stormConf, context, collector);
     this.collector = collector;
+    if(getSensorParserConfig() == null) {
+      filter = new GenericMessageFilter(null);
+    }
+    else if(filter == null) {
+      filter = Filters.get(getSensorParserConfig().getFilterClassName()
+              , getSensorParserConfig().getParserConfig()
+      );
+    }
     parser.init();
     writer.init();
   }
 
+  protected SensorParserConfig getSensorParserConfig() {
```

Comment:
Would it make more sense to move this to the ConfiguredParserBolt class?  Don't think it matters much now because ParserBolt is the only class that extends ConfiguredParserBolt but other bolts may be created in the future and it would be nice to have this method available.